// vue-text-to-speech/webpack.config.js
const webpack = require('webpack');
const merge = require('webpack-merge');
const path = require('path');

var config = {
  output: {
    path: path.resolve(__dirname + '/dist/'),
  },
  module: {
    loaders: [
      {
        test: /\.js$/,
        loader: 'babel',
        include: __dirname,
        exclude: /node_modules/
      },
      {
        test: /\.vue$/,
        loader: 'vue'
      },
      {
        test: /\.css$/,
        loader: 'style!less!css'
      }
    ]
  },
  /*externals: {
    moment: 'moment'
  },*/
  plugins: [
    new webpack.optimize.UglifyJsPlugin( {
      minimize : true,
      sourceMap : false,
      mangle: true,
      compress: {
        warnings: false
      }
    } )
  ]
};


module.exports = [
  merge(config, {
    entry: path.resolve(__dirname + '/src/plugin.js'),
    output: {
      filename: 'vue-text-to-speech.min.js',
      libraryTarget: 'window',
      library: 'VueTextToSpeech',
    }
  }),
  merge(config, {
    entry: path.resolve(__dirname + '/src/speech.vue'),
    output: {
      filename: 'vue-text-to-speech.js',
      libraryTarget: 'umd',
      library: 'vue-text-to-speech',
      umdNamedDefine: true
    }
  })
];
